pulsar-broker-common/src/main/java/com/yahoo/pulsar/broker/ServiceConfiguration.java

```diff
@@ -258,6 +262,16 @@
     // Name of load manager to use
     @FieldContext(dynamic = true)
     private String loadManagerClassName = "com.yahoo.pulsar.broker.loadbalance.impl.SimpleLoadManagerImpl";
+    @FieldContext(dynamic = true)
+    private String activeVersion = "";
+    // Name of the Maven resource which contains the version string and build timestamp
+    private final String RESOURCE_NAME = "version.txt";
+    // Version string for this broker (inferred from Maven artifact, overridden in the broker configuration file)
+    private String brokerVersionString;
+    // Build timestamp
+    private String brokerBuildTimeString;
+    // Start time (seconds since 1/1/1970 UTC) of this broker
+    private long brokerStartTime = System.currentTimeMillis()/1000;
```

Comment:
I would prefer to keep the auto-generated version string and timestamp outside of the configuration object. Maybe something like `PulsarService.getBrokerVersion()`?